custom/abt/reports/tests/test_flagspecs.py

```diff
@@ -0,0 +1,16 @@
+from glob import glob
+
+from yaml import load, Loader, parser
+
+
+def test_yaml_formatting():
+    for filename in glob('custom/abt/reports/*.yaml'):
+        yield check_yaml_file, filename
```

Comment:
is this a fancy way of authoring tests that I'm not familiar with? The test runner somehow resolves this to `check_yaml_file(filename)`?

Reply:
> The test runner somehow resolves this to `check_yaml_file(filename)` at the right time?

Yeah, exactly.

nose (and pytest) support [test generators](https://docs.nose2.io/en/latest/plugins/generators.html). I went this route for two reasons:

1. If one file fails, the test runner will keep going, so you can see all the files that have problems instead of just the first one.
2. The test runner makes this look like a test for each file, instead of a single test. (It just looks nicer when your files pass.)